backend/src/index.ts

```diff
@@ -106,10 +141,12 @@ const processSourceBlocks = (target: Target) => async (source: Source) => {
       archives.forEach(async archive => {
         let lastBlockProcessingReportAt = Date.now();
         let processedBlocks = 0;
-
+        const q = new AsyncQueue(500);
         let nonce = (await target.api.rpc.system.accountNextIndex((archive.signer as KeyringPair).address)).toBn();
+
         for await (const blockData of archive.getBlocks()) {
-          target.sendBlockTx(blockData, nonce);
+          q.add(() => target.sendBlockTx({ ...blockData, nonce }));
```

Comment:
This is a good way to do OOM, we need to `await` here or else we will quickly run out of memory (remember, we can read ~5000 blocks/s, but only send ~500 blocks/s to the blockchain, so the queue will grow until it comsumes all available memory).

Reply:
then do we need `waiting` array at all or should we set a limit on it?

Reply:
We need to wait if we hit a limit and only until there is an empty slot.

I'm actually changing code to use `batchAll` call from pallet `utility`, so this PR might become unnecessary. Please hold on tweaking it before I have another PR ready so we don't waste an effort here.

Reply:
sure, please, let me know

Reply:
With #37 this is no longer needed for archives, but is still potentially relevant for regular mode

Reply:
I mean there is no reason to implement a special queue for rx.js, cause it has its own tools (I assume they have some kind of queue under the hood). So I am wondering if we need this for regular mode how it should look like

Reply:
If you can write necessary logic with rx.js, why not...